### PR TITLE
Flags to support embedded Linux

### PR DESCRIPTION
**Pull request by jwinarske**
- FILAMENT_SKIP_SDL2=ON is used to  skip dependencies of sdl2 and sdl2 CMake
    
- FILAMENT_LINUX_IS_MOBILE is used for Mobile build

Discussion thread here: https://github.com/google/filament/issues/4874

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,10 @@ option(FILAMENT_SUPPORTS_XLIB "Include XLIB support in Linux builds" ON)
 
 option(FILAMENT_SUPPORTS_WAYLAND "Include Wayland support in Linux builds" OFF)
 
+option(FILAMENT_SKIP_SDL2 "Skip dependencues of SDL2, and SDL2" OFF)
```
Comment (romainguy) [inline CMakeLists.txt:28]: Typo in `dependencues`

**Comment by jwinarske** [comment]
Reference implementation here:  https://github.com/jwinarske/meta-vulkan/tree/main/recipes-graphics/filament

File: libs/bluevk/CMakeLists.txt [hunk 2]
```diff
@@ -28,7 +28,7 @@ target_link_libraries(${TARGET} utils math)
 target_include_directories(${TARGET} PUBLIC ${PUBLIC_HDR_DIR})
 
 # test_bluevk is not supported on mobile or Windows
-if (NOT ANDROID AND NOT IOS AND NOT WIN32)
+if (NOT ANDROID AND NOT IOS AND NOT WIN32 AND NOT FILAMENT_SKIP_SDL2)
```
Comment (romainguy) [inline libs/bluevk/CMakeLists.txt:31]: Don't you need to also skip libapp and the native samples when FILAMENT_SKIP_SDL2 is set?

Comment (jwinarske) [reply libs/bluevk/CMakeLists.txt:31]: samples only build in case of host

File: CMakeLists.txt (hunk 1 above)
Comment (jwinarske) [reply CMakeLists.txt]: done

File: libs/bluevk/CMakeLists.txt (hunk 2 above)
Comment (jwinarske) [reply libs/bluevk/CMakeLists.txt:31]: The two use cases are tools and target -dev package build.

```
EXTRA_OECMAKE:class-native += " \
    -D BUILD_SHARED_LIBS=OFF \
    -D CMAKE_BUILD_TYPE=Release \
    -D FILAMENT_SUPPORTS_VULKAN=ON \
    -D FILAMENT_SUPPORTS_OPENGL=OFF \
    -D FILAMENT_BUILD_FILAMAT=OFF \
    -D FILAMENT_SKIP_SAMPLES=ON \
    -D FILAMENT_SKIP_SDL2=ON \
    ${PACKAGECONFIG_CONFARGS} \
    "

EXTRA_OECMAKE:class-target += " \
    -D BUILD_SHARED_LIBS=OFF \
    -D CMAKE_BUILD_TYPE=Release \
    -D FILAMENT_SUPPORTS_VULKAN=ON \
    -D FILAMENT_SUPPORTS_OPENGL=OFF \
    -D FILAMENT_LINUX_IS_MOBILE=ON \
    -D IMPORT_EXECUTABLES_DIR=. \
    -D FILAMENT_SKIP_SDL2=ON \
    -D DIST_ARCH=${BUILD_ARCH} \
    -D FILAMENT_HOST_TOOLS_ROOT=${STAGING_BINDIR_NATIVE} \
    ${PACKAGECONFIG_CONFARGS} \
    "
```

**Review by romainguy — APPROVED** [review]

**Pull request merged**